### DISCUSSION TITLE
Transparent PNG is losing color when using composite

### DISCUSSION THREAD

**Discussion by bullmoose20** [ImageMagick/ImageMagick#5708] [Help] [answered]
This is the transparent image:
![resized_OG](https://user-images.githubusercontent.com/12549033/199332170-401d1008-8e7c-4e80-a26a-d4756ffe3049.png)

Here is the background:
![base_#575757](https://user-images.githubusercontent.com/12549033/199331558-d00820d2-f34f-437a-bf4c-8ad1a4fe7f74.jpg)

When I run the following command: `magick composite -gravity center -geometry +0+0 resized_OG.png tmp\base_#575757.jpg tmp\transparent.jpg` I get this:
![transparent](https://user-images.githubusercontent.com/12549033/199332377-0a33e969-4c39-46c2-9856-40a3fa17ffa6.jpg)

I would like to have the colors remain in the logo and not sure why this is happening.

**Comment by bullmoose20** [comment, marked as answer]
`magick composite -gravity center -geometry +0+0 -colorspace sRGB resized_OG.png tmp\base_#575757.jpg tmp\transparent.jpg` produced this:
![transparent](https://user-images.githubusercontent.com/12549033/199340793-d45b7784-7851-420e-ab23-a356bc9c12e5.jpg)

The key was the `-colorspace sRGB`

**Comment by snibgo** [comment]
You fixed it. It is happening because your background image is grayscale. This metadata is carried forwards to the output. Set that input to sRGB to fix that problem.

Also, I suggest not using "magick composite". Instead, use just "magick", with the "-composite" operation. This needs the input files in the opposite order. So:
```
magick tmp\base_#575757.jpg -set colorspace sRGB resized_OG.png -gravity center -geometry +0+0 -composite tmp\transparent.jpg
```

**Reply by bullmoose20** [reply]
thanks!
